microsoft/onnxruntime
# [v2] Add ACL (Arm Compute Library) execution provider

Add a new execution provider targeting Arm architecture based on Arm Compute Library.
Validated on NXP i.MX8QM CPU with ResNet50, MobileNetv2 and VGG models.
All unit tests are passing.

Comparative performance improvements for ResNet50v1 model obtained with
onnxruntime_perf_test:
		A72	2xA72	A53	4xA53
ACL vs CPU  	16%	9%	21%	13%

Usage documentation available in ACL-ExecutionProvider.

v5:
- Remove Memcpy node and kernel dependencies
- Move prints to verbose log
- Revert onnx external version change 
- Run clang-format

v4:
- Revert OPENMP multicore fix, no longer necessary
- Default to ACL 19.05

v3:
- Removed unused InsertFusedRules() function
- Removed redundant acl_provider_factory.h
- Moved createMemoryManager() function to acl_common.cc
- Enabled Pooling with ACL 
- Fixed tests feedback: enable_cpu_mem_arena initialization, ORT_UNUSED_PARAMETER

v2:
- integrated perf test
- integrated unit tests
- fixed CONV unit tests
- added documentation
